nexusparser/tools/dataconverter/helpers.py

```diff
@@ -67,6 +75,10 @@ def generate_template_from_nxdl(root, template, path=None, nxdl_root=None):
         if tag == "field" \
            and ("units" in root.attrib.keys() and root.attrib["units"] != "NX_UNITLESS"):
             template[optionality][f"{path}/@units"] = None
+    elif tag == "group":
+        if is_a_lone_group(root):
+            template[get_required_string(root)][path] = None
+            template["lone_groups"] = path
```

Comment:
I think this needs an array here to also reflect multiple empty and required groups. See failing test.